source/runbooks/revoke-network-access.html.md.erb

```diff
@@ -0,0 +1,96 @@
+---
+owner_slack: "#modernisation-platform"
+title: Revoke network access if required
+last_reviewed_on: 2024-03-04
+review_in: 6 months or when is resolved.
+---
+
+# <%= current_page.data.title %>
+
+# Summary
+
+The steps listed deny access on a platform level. If additional changes are required at an application level then these must be undertaken by the application owning team.
+
+A network attack is an exceptional situation so quick amendments are needed. It is anticiapted that this can be done quickley through the console.
+
+This document lists the steps needed to stop access to and from the internet and prevening the system from being accessed from or accessing other applications on the MoJ.
+
+All changes are to be made manually on the relevant environment (for example core-vpc-production) and are not made through the code. If a way of reverting back to baseline is required the terraform code can be run and this will reset everything.
+
+## Stopping access from the internet to an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+If you know which VPC application area is being attacked you can remove it from the VPC.
+
+1. Go to VPC->Internet gateways
+2. Select the gateway (example is hmpps-preproduction-internet-gateway)
+3. Note the Internet gateway ID and the VPC ID
+4. Click on Actions, Detach from VPC
+5. If you are sure this is correct click on the Detach internet gateway button
+
+The selected internet gateway will be detached from the VPC.
+
+This change will affect all applications in the VPC that you have selected. No applications in the VPC will be accessible from the internet.
+
+## Stopping access to the internet for an application on MP
+
+This work is carried out from the core-vpc-(environment) environment e.g. core-vpc-production
+
+The process needs to be done for all 4 regions listed. An example is
+
+- hmpps-production-data-nacl
+- hmpps-production-private-nacl
+- hmpps-production-protected-nacl
+- hmpps-production-public-nacl
+
+Follow the process below for each of these
+
+1. Go to VPC->Security->Network ACLs
+2. Click on the checkbox for the one you want to change then select **outbound rules**.
+3. Due to the way checks work in AWS (see [here](https://docs.aws.amazon.com/network-firewall/latest/developerguide/suricata-rule-evaluation-order.html) ) we need to amend each one in place to Deny where it is currently Allow. **Note** any that are currently set to Deny in case this is manually changed back.
```

Comment:
This is not true and the and the linked documentation is irrelevant here, as it is related to the firewall changes and not NACLs.
See [this documentation ](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-network-acls.html) on the evaluation order of the rules and update the documentation to reflect it.
Think about a smarter way to implement it given the rule evaluation behaviour, updating every single rule in each NACL sounds like a lot of manual work.

Reply:
No more than 10 rules to amend in each. Sounds like a lot but changing from allow to deny isn't too much effort

Reply:
See my previous comment, as what you are saying about the NACLs rule evaluation is incorrect.

Reply:
We have reviewed this again and have amended the documentation to add a new line to the rules to add a global deny. I have listed the current section as an option.